.github/workflows/ansible-test.yml

```diff
@@ -3,23 +3,84 @@ name: CI
 on: push
 
 jobs:
-  build:
+  sanity:
     runs-on: ubuntu-latest
+    #services:
+    #  grafana:
+    #    image: grafana/grafana:latest
+    #    ports:
+    #    - 3000/tcp
     steps:
+
+      - name: Check out code
+        uses: actions/checkout@v1
+        with:
+          path: ansible_collections/gundalow_collections/grafana
+
+      - name: Set up Python 3.6
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.6
+
+      - name: Install Ansible 2.9.1
+        run: pip install ansible==2.9.1 --disable-pip-version-check
+
+      - name: Run sanity tests
+        run: ansible-test sanity --docker -v --color
+
+  units:
+    runs-on: ubuntu-latest
+    #services:
+    #  grafana:
+    #    image: grafana/grafana:latest
+    #    ports:
+    #    - 3000/tcp
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v1
+        with:
+          path: ansible_collections/gundalow_collections/grafana
+
       - name: Set up Python 3.6
         uses: actions/setup-python@v1
         with:
           python-version: 3.6
+
+      - name: Install Ansible 2.9.1
+        run: pip install ansible==2.9.1 --disable-pip-version-check
+
+      - name: Run unit tests
+        run: ansible-test units --docker -v --color
+
+  integration:
+    runs-on: ubuntu-latest
+    #services:
+    #  grafana:
+    #    image: grafana/grafana:latest
+    #    ports:
+    #        - 3000/tcp
+    strategy:
+      matrix:
+        python_version: ["3.5", "3.6"]
+    steps:
       - name: Check out code
         uses: actions/checkout@v1
         with:
           path: ansible_collections/gundalow_collections/grafana
+
+      - name: Set up Python ${{ matrix.python_version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python_version }}
+
       - name: Install Ansible 2.9.1
         run: pip install ansible==2.9.1 --disable-pip-version-check
-      - name: Run sanity tests
-        run: ansible-test sanity --docker -v --color
-      - name: Run integration tests on Python 3.6
-        run: ansible-test integration --docker ubuntu1804 -v --color --retry-on-error --python 3.6
+
+      - name: Run integration tests on Python ${{ matrix.python_version }}
+        run: ansible-test integration --docker -v --color --retry-on-error --python ${{ matrix.python_version }}
+        env:
```

Comment:
`ansible-test` by design ignores any `env`, perhaps you could write these values into a `vars` file in the role that the tasks would see?

Reply:
I can't.
The grafana URL will be dynamic since I use the `services` feature from github actions.
The port mapping will be different from a tests to another.

I wrote you a small email about that.